examples/s3_ingest.py

```diff
@@ -38,8 +38,10 @@
 ds = (
     ctx.read.binary(paths, binary_format="pdf", filesystem=fsys)
     .partition(partitioner=UnstructuredPdfPartitioner())
-    .merge(merger=GreedyTextElementMerger(tokenizer=tokenizer, max_tokens=512))
     .extract_entity(entity_extractor=OpenAIEntityExtractor("title", llm=davinci_llm, prompt_template=title_template))
+    .merge(merger=GreedyTextElementMerger(tokenizer=tokenizer, max_tokens=512))
+    .spread_properties(["path", "title"])
+    .split_elements(tokenizer=tokenizer, max_tokens=512)
```

Comment:
I'm wondering if this should just be rolled into the `merge` transform and have it merge without creating too large chunks.  Otherwise, one has to remember to call `split_elements` after doing `merge`.

Reply:
You're not wrong.  The merge and split are dirty laundry to a certain extent.  I think Vinayak's idea of a chunker that is passed to the partitioner would clean this up a bit.  Another way to go would be to have the embedder do the chunking it needs.  For power-users, it may still be useful to have top-level ad-hoc merge and split operators.